detox/jest.detox.startup.js

```diff
@@ -69,3 +83,30 @@ async function prepDeveloperApp(url, port) {
     await element(by.id("text_input_runtime_url")).tapReturnKey();
     await device.setURLBlacklist([]);
 }
+
+async function setDemoMode() {
+    if (device.getPlatform() === "ios") {
+        const type = device.name.substring(device.name.indexOf("(") + 1, device.name.lastIndexOf(")"));
+        execSync(
+            `xcrun simctl status_bar "${type}" override --time "12:00" --batteryState charged --batteryLevel 100 --wifiBars 3 --cellularMode active --cellularBars 4`
+        );
+    } else {
+        const id = device.id;
+        // enter demo mode
+        execSync(`adb -s ${id} shell settings put global sysui_demo_allowed 1`);
+        // display time 12:00
+        execSync(`adb -s ${id} shell am broadcast -a com.android.systemui.demo -e command clock -e hhmm 1200`);
+        // Display full mobile data with 4g type and no wifi
+        execSync(
+            `adb -s ${id} shell am broadcast -a com.android.systemui.demo -e command network -e mobile show -e level 4 -e datatype 4g -e wifi false`
+        );
+        // Hide notifications
+        execSync(
+            `adb -s ${id} shell am broadcast -a com.android.systemui.demo -e command notifications -e visible false`
```

Comment:
Does this disable the device's native notifications capabilities? What about testing notifications widgets/actions?

Reply:
Potentially yes. We could add this to execute before every screenshot and exit demo again after screenshot. However, current solution works in 99% of the cases. Lets worry about notifications when we get there.

Reply:
Agreed.